src/main/kotlin/com/meltwater/docker/compose/DockerCompose.kt

```diff
@@ -11,61 +12,108 @@ import org.slf4j.LoggerFactory
-            val tmpFile = File.createTempFile(prefix, ".yml")
-            tmpFile.writeBytes(IOUtils.toString(ymlStream, Charset.forName("utf-8")).toByteArray())
-            ymlStream.close()
-            return tmpFile.path
-        } catch (ex: Exception) {
-            throw RuntimeException("Could not load yml file from classpath: $classPathYmlResource", ex)
+        private var MIN_DOCKER_COMPOSE_VERSION: DefaultArtifactVersion = DefaultArtifactVersion("1.5.0")
+
+
+        private const val YAML_FILE_EXTENSION = ".yml"
+
+        private const val ENVIRONMENT_FILE_EXTENSION = ".env"
+
+        private val ENVIRONMENT_VARIABLE_SEPARATOR = Pattern.compile("=")
+
+        private fun initializeNewDockerComposeSession(classPathYmlResource: String, env: Map<String, String>): String {
+            val composeFile = File.createTempFile(classPathYmlResource, YAML_FILE_EXTENSION)
+            val dockerComposeFilename = composeFile.absolutePath
+            saveResourceToTmpFolder(classPathYmlResource, composeFile)
+            if (!composeFile.exists()) {
+                throw RuntimeException("File $dockerComposeFilename could not be found.")
+            }
+            val environmentFileName = dockerComposeFilename.substringBeforeLast(YAML_FILE_EXTENSION) + ENVIRONMENT_FILE_EXTENSION
+            saveEnvironmentFile(env, environmentFileName)
+            return dockerComposeFilename
+        }
+
+        private fun saveResourceToTmpFolder(classPathYmlResource: String, destinationFile: File) {
+            try {
+                val ymlStream: InputStream = DockerCompose::class.java.classLoader.getResourceAsStream(classPathYmlResource)
+                val yamlContents = IOUtils.toString(ymlStream, Charset.forName("utf-8"))
+                destinationFile.writeBytes(yamlContents.toByteArray())
+                ymlStream.close()
+            } catch (ex: Exception) {
+                throw RuntimeException("Could not load yml file from classpath: $classPathYmlResource", ex)
+            }
         }
+
+        private fun saveEnvironmentFile(env: Map<String, String>, environmentFileName: String) {
+            val environmentPath = Paths.get(environmentFileName)
+            val environment = env.entries.stream().map { "${it.key}=${it.value}" }.collect(Collectors.joining("\n"))
+            Files.write(environmentPath, environment.toByteArray(), StandardOpenOption.TRUNCATE_EXISTING, StandardOpenOption.WRITE, StandardOpenOption.CREATE)
+        }
+
+        private fun loadEnvironmentFile(fileNamePrefix: String): Map<String, String> {
+            val environmentPath = Paths.get(fileNamePrefix + ENVIRONMENT_FILE_EXTENSION)
+            if (!(Files.exists(environmentPath) && Files.isRegularFile(environmentPath))) {
+                LOGGER.warn("Cannot find the environment file: $environmentPath")
+                return emptyMap()
+            }
+            return Files.lines(environmentPath)
```

Comment:
Same here, should be a kotlin equivalent

Reply:
Sure? Other than the first line, it is a simple map/filter/reduce chain. I don't expect it would be much different in Kotlin?

Reply:
I found [https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.io/java.io.-file/read-lines.html](this), however, since it produces an eager collection, I don't think it would be worth converting to it.

Reply:
Ok. Leave as is then